engine/execution/computation/computer/computer.go

```diff
@@ -96,111 +106,126 @@ func (e *blockComputer) ExecuteBlock(
 }
 
 func (e *blockComputer) executeBlock(
-	ctx context.Context,
+	blockSpan opentracing.Span,
 	block *entity.ExecutableBlock,
 	stateView state.View,
 	programs *programs.Programs,
 ) (*execution.ComputationResult, error) {
 
 	blockCtx := fvm.NewContextFromParent(e.vmCtx, fvm.WithBlockHeader(block.Block.Header))
-
 	collections := block.Collections()
-
-	var gasUsed uint64
-
-	interactions := make([]*delta.SpockSnapshot, len(collections)+1)
-
-	events := make([]flow.Event, 0)
-	serviceEvents := make([]flow.Event, 0)
-	blockTxResults := make([]flow.TransactionResult, 0)
+	res := &execution.ComputationResult{
+		ExecutableBlock:    block,
+		Events:             make([]flow.Event, 0),
+		ServiceEvents:      make([]flow.Event, 0),
+		TransactionResults: make([]flow.TransactionResult, 0),
+		StateCommitments:   make([]flow.StateCommitment, 0),
+		Proofs:             make([][]byte, 0),
+	}
 
 	var txIndex uint32
+	var err error
+	var wg sync.WaitGroup
+
+	stateCommitments := make([]flow.StateCommitment, 0, len(collections)+1)
+	proofs := make([][]byte, 0, len(collections)+1)
+
+	bc := blockCommitter{
+		committer: e.committer,
+		blockSpan: blockSpan,
+		tracer:    e.tracer,
+		state:     block.StartState,
+		views:     make(chan state.View, len(collections)+1),
+		callBack: func(state flow.StateCommitment, proof []byte, err error) {
+			defer wg.Done()
+			if err != nil {
+				panic(err)
```

Comment:
just checking we want to panic here?

Reply:
yeah, we should panic if we can't commit the deltas.